.github/workflows/live-test.yml

```diff
@@ -1,24 +1,61 @@
 name: Live tests for B2K
 on:
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
+  repository_dispatch:
+    types: [ok-to-test-command]
+permissions:
+  id-token: write
 jobs:
   liveTest:
+    env:
+      PR_NUMBER: ${{ github.event.client_payload.slash_command.args.named.PR_NUMBER }}
+      HEAD_REPO: ${{ github.event.client_payload.slash_command.args.named.HEAD_REPO }}
+      HEAD_BRANCH: ${{ github.event.client_payload.slash_command.args.named.HEAD_BRANCH }}
     strategy:
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     name: live test for b2k
     steps:
     - uses: actions/checkout@v2
+      # checkout PR head repo/branch
+      with:
+         repository: ${{ env.HEAD_REPO }}
+         ref: ${{ env.HEAD_BRANCH }}
+    - name: Login Azure Context
+      if: ${{ matrix.os != 'ubuntu-latest' }}
+      uses: Azure/login@v1
+      with:
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+    - name: Prepare Cluster name Mac and Windows
+      if: ${{ matrix.os == 'macos-latest' ||  matrix.os == 'windows-latest' }}
+      shell: bash
+      run: |
+        if [ "$RUNNER_OS" == "Windows" ]; then
+          echo "cluster_name=b2k-livetest-win-pr-$PR_NUMBER" >> $GITHUB_ENV
+        else
+          echo "cluster_name=b2k-livetest-mac-pr-$PR_NUMBER" >> $GITHUB_ENV
+        fi
```

Comment:
@hsubramanianaks thanks for suggestion. Combined AKS steps but still need to do this here because "windows" is not allowed as either name or substring FYI.

Reply:
Yep agree, cluster name is required in the environment variables since it is used in subsequent steps.